tests/model_registry/model_catalog/test_default_model_catalog.py

```diff
@@ -152,28 +153,89 @@ def test_model_default_catalog_get_model_artifact(
         assert result, f"No artifacts found for {model_name}"
         assert result[0]["uri"]
 
+
+@pytest.mark.skip_must_gather
+class TestModelCatalogDefaultData:
+    """Test class for validating default catalog data (not user-specific)"""
+
     def test_model_default_catalog_number_of_models(
         self: Self,
-        admin_client: DynamicClient,
-        model_registry_namespace: str,
-        model_catalog_rest_url: list[str],
-        user_token_for_api_calls: str,
+        default_catalog_api_response: dict[Any, Any],
+        default_model_catalog_yaml_content: dict[Any, Any],
     ):
         """
         RHOAIENG-33667: Validate number of models in default catalog
         """
 
-        model_catalog_pod = get_model_catalog_pod(
-            client=admin_client, model_registry_namespace=model_registry_namespace
-        )[0]
+        count = len(default_model_catalog_yaml_content.get("models", []))
 
-        catalog_content = model_catalog_pod.execute(command=["cat", DEFAULT_CATALOG_FILE], container=CATALOG_CONTAINER)
-        catalog_data = yaml.safe_load(catalog_content)
-        count = len(catalog_data.get("models", []))
+        assert count == default_catalog_api_response["size"], (
+            f"Expected count: {count}, Actual size: {default_catalog_api_response['size']}"
+        )
+        LOGGER.info("Model count matches")
 
-        result = execute_get_command(
-            url=f"{model_catalog_rest_url[0]}models?source={DEFAULT_CATALOG_ID}&pageSize=1",
-            headers=get_rest_headers(token=user_token_for_api_calls),
+    def test_model_default_catalog_correspondence_of_model_name(
+        self: Self,
+        default_catalog_api_response: dict[Any, Any],
+        default_model_catalog_yaml_content: dict[Any, Any],
+    ):
+        """
+        RHOAIENG-35260: Validate the correspondence of model parameters in default catalog yaml and model catalog api
+        """
+
+        api_models = {model["name"]: model for model in default_catalog_api_response.get("items", [])}
+
+        models_with_differences = {}
+
+        for model in default_model_catalog_yaml_content.get("models", []):
+            LOGGER.info(f"Validating model: {model['name']}")
+
+            api_model = api_models.get(model["name"])
+
+            # Exclude artifacts and null-valued properties from YAML model comparison
+            model_filtered = {k: v for k, v in model.items() if k != "artifacts" and v is not None}
```

Comment:
I don't love this, IMHO we shouldn't be expecting differences and here instead we are filtering out the "problematic" fields. I would at least log a warning with the details of what we discard.

Reply:
Agree, I opened a bug for it https://issues.redhat.com/browse/RHOAIENG-35322